tile/targets/cpu/compiler.cc

```diff
@@ -1846,16 +1846,47 @@ void Compiler::OutputBool(llvm::Value* ret, const stripe::Intrinsic& intrinsic)
   ret->setName(intrinsic.outputs[0]);
 }
 
-void Compiler::CallIntrinsicFunc(const stripe::Intrinsic& stmt, const char* name_f32, const char* name_f64) {
-  assert(1 == stmt.inputs.size());
-  Scalar op = Cast(scalars_[stmt.inputs[0]], stmt.type);
-  std::vector<llvm::Value*> argvals{op.value};
+void Compiler::CallIntrinsicFunc(const stripe::Intrinsic& stmt, const char* name_f32, const char* name_f64,
+                                 const size_t numParamsIn) {
+  size_t numParams = stmt.inputs.size();
+  switch (numParamsIn) {
+    case 1:
+      if (numParams != 1) {
+        throw std::runtime_error("CallIntrinsicFunction expects 1 parameter");
+      }
+      break;
+    case 2:
+      if (numParams != 2) {
+        throw std::runtime_error("CallIntrinsicFunction expects 2 parameters");
+      }
+      break;
+    default:
+      throw std::runtime_error("CallIntrinsicFunction expects 1 or 2 parameters");
+  }
```

Comment:
Seems like this whole thing could be something like: 
```
if (stmt.inputs.size() != numParamsIn) {
  throw std::runtime_error(llvm::formatv("CallIntrinsicFunction expects {0} parameter(s), actual: {1}", numParamsIn, stmt.inputs.size());
}

Reply:
There is handling of 2 params only. In case numParamsIn and stmt.inputs.size() are 3 for example, the code will still not work properly.